### PR TITLE
Batched cls avg

### PR DESCRIPTION
**Pull request by garrettwrong**
Depends on #686 .

~WIP~

### CODE REVIEW

**Comment by garrettwrong** [comment]
Docs looked okay (except for unrelated expansion bug fixed upstream).

Reviewed `simulated_abinitio_pipeline.py` experiment (current time 15mins on laptop w/ 8 cores and 64GB ram).

File: src/aspire/basis/fspca.py [hunk 1]
```diff
@@ -283,6 +229,87 @@ def _compute_spca(self, coef):
...
+            # #  then use the transpose so image stack becomes columns.
+
+            # Initialize a totally empty BlkDiagMatrix, then build incrementally.
+            A = BlkDiagMatrix.empty(0, dtype=batch_coef.dtype)
+
+            # Zero angular index is special case of indexing.
+            A_0 = batch_coef[:, zero_ell_mask] - self.mean_coef_zero
+            A.append(A_0)
+
+            # Remaining angular indices have postive and negative entries in real representation.
```
Comment (chris-langfield) [inline src/aspire/basis/fspca.py:270]: test comment

**Review by chris-langfield — DISMISSED** [review]

**Comment by garrettwrong** [comment]
Updated to latest develop. Checked simulated pipeline and class average gallery example.

**Comment by garrettwrong** [comment]
Merged develop and small update after #681  .

File: src/aspire/basis/fspca.py [hunk 2]
```diff
@@ -282,6 +229,86 @@ def _compute_spca(self, coef):
...
+            for ell in range(
+                1, self.basis.ell_max + 1
+            ):  # `ell` in this code is `k` from paper
+                mask = self.basis._indices["ells"] == ell
+                mask_pos = [
```
Comment (janden) [inline src/aspire/basis/fspca.py]: `mask_pos = mask & (self.basis._indices["sgns"] == +1)`?

File: src/aspire/basis/fspca.py [hunk 3]
```diff
@@ -45,6 +44,7 @@ def __init__(
         Default value of `None` will estimate noise with WhiteNoiseEstimator.
         Use 0 when using clean images so cov2d skips applying noisy covar coeffs..
         :param batch_size: Batch size for computing basis coefficients.
+        `batch_size` is also passed to BatchedRotCov2D.
```
Comment (janden) [inline src/aspire/basis/fspca.py]: Indent (see #704)?

**Review by janden — CHANGES_REQUESTED** [review]
Looks great. Just two things.

File: src/aspire/basis/fspca.py (hunk 3 above)
Comment (garrettwrong) [reply src/aspire/basis/fspca.py]: changed.

File: src/aspire/basis/fspca.py (hunk 2 above)
Comment (garrettwrong) [reply src/aspire/basis/fspca.py]: Yeah, I think that is better.  Changed.

Sorry, I lifted that directly from `covar2d.py` without too much thought. I'll make a separate pr for other one(s).

Thanks!

**Review by janden — APPROVED** [review]

**Pull request merged**